cve_bin_tool/output_engine/__init__.py

```diff
@@ -594,10 +594,50 @@ def output_pdf(
         exploits: bool = False,
         all_product_data=None,
     ):
+        """Output a PDF of CVEs
+        Required module: Reportlab not found"""
         LOGGER.warning("PDF output requires install of reportlab")
 
 
 class OutputEngine:
+    """
+    Class represention of OutputEngine
+    Attributes:
+        all_cve_data (dict[ProductInfo, CVEData])
+        scanned_dir (str)
+        filename (str)
+        themes_dir (str)
+        time_of_last_update
+        tag (str)
+        logger
+        products_with_cve (int)
+        products_without_cve (int)
+        total_files (int)
+        is_report (bool)
+        no_zero_cve_report (bool)
+        append
+        merge_report
+        affected_versions (int)
+        all_cve_version_info
+        detailed (bool)
+        vex_filename (str)
+        exploits (bool)
+        all_product_data
+        sbom_filename (str)
+        sbom_type (str)
+        sbom_format (str)
+        sbom_root (str)
+        offline (bool)
+
+    Methods:
+        output_cves: Output a list of CVEs to other formats like CSV or JSON.
+        generate_vex: Generate a vex file and create vulnerability entry.
+        generate_sbom: Create SBOM package and generate SBOM file.
+        output_file_wrapper:
+        output_file: Generate a file for list of CVE
+        check_file_path: Generate a new filename if file already exists.
+        check_dir_path: Generate a new filename if filepath is a directory.
+    """
```

Comment:
```suggestion
    """

```
Black wants an extra blank line here.